src/components/common/Directors.vue

```diff
@@ -831,98 +786,105 @@ export default class Directors extends Mixins(CommonMixin, DateMixin, DirectorMi
+      postalCode: address.postalCode || '',
+      streetAddress: address.streetAddress || '',
+      streetAddressAdditional: address.streetAddressAdditional || ''
     }
   }
 
-  /**
-   * Function called externally to set the draft date.
-   * FUTURE: change this to a prop
-   * @param date The draft date to set.
-   */
-  public setDraftDate (date): void {
-    this.draftDate = date
-  }
-
-  /**
-   * Function called internally and externally to fetch the list of directors.
-   */
-  public async getDirectors (getOrigOnly: boolean = false): Promise<void> {
+  /** Fetches the list of directors on As Of Date from the Legal API. */
+  // FUTURE: this API call should be in the parent component or some mixin/service
+  private async fetchDirectors (): Promise<void> {
     if (this.entityIncNo && this.asOfDate) {
       const url = `businesses/${this.entityIncNo}/directors?date=${this.asOfDate}`
-      await axios.get(url)
-        .then(response => {
-          if (response && response.data && response.data.directors) {
-            // note - director list manipulated locally here (attributes added), THEN saved to this.directors,
-            // otherwise new attributes are not reflected in initial draw of HTML list.
-
-            let directors = response.data.directors as DirectorIF[]
-
-            directors = directors.sort(this.fieldSorter(['lastName', 'firstName', 'middleName']))
-
-            for (let i = 0; i < directors.length; i++) {
-              directors[i].id = i + 1
-              directors[i].isFeeApplied = (directors[i].isFeeApplied !== undefined) ? directors[i].isFeeApplied : false
-              directors[i].isDirectorActionable = (directors[i].cessationDate === null)
-              directors[i].actions = []
-
-              // if there is no officer middle initial field, add it with blank data
-              if (!directors[i].officer.hasOwnProperty('middleInitial')) directors[i].officer.middleInitial = ''
-
-              // save previous officer name data for COLIN to use when updating record
-              directors[i].officer.prevFirstName = directors[i].officer.firstName
-              directors[i].officer.prevLastName = directors[i].officer.lastName
-              directors[i].officer.prevMiddleInitial = directors[i].officer.middleInitial
-
-              // ensure there is complete address data including missing/blank fields
-              directors[i].deliveryAddress = this.formatAddress(directors[i].deliveryAddress)
-              if (directors[i].mailingAddress) {
-                directors[i].mailingAddress = this.formatAddress(directors[i].mailingAddress)
-              }
+      await axios.get(url).then(response => {
+        if (response?.data?.directors) {
+          const directors = response.data.directors as DirectorIF[]
+
+          this.original = directors.sort(this.fieldSorter(['lastName', 'firstName', 'middleName']))
+
+          this.original.forEach((director, i) => {
```

Comment:
Simple change from for loop to forEach loop. (Ignore whitespace changes to see this better.)